nats/nats_test.go

```diff
@@ -0,0 +1,269 @@
+package nats
+
+import (
+	"testing"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+	"github.com/stretchr/testify/require"
+)
+
+var config = Config{
+	URLs: "nats://localhost:4443",
+	NatsOptions: []nats.Option{
+		nats.MaxReconnects(2),
+		// Enable TLS by specifying RootCAs
+		nats.RootCAs("../tls/ca.crt"),
```

Comment:
This is missing the following:

```go
nats.ClientCert("client-cert.pem", "client-key.pem")
```

Source https://docs.nats.io/using-nats/developer/connecting/tls#connecting-with-tls-and-verify-client-identity

Reply:
These are the generated paths during CI:

- /home/runner/work/storage/storage/tls/client.crt
- /home/runner/work/storage/storage/tls/client.key

Reply:
Just by specifying `nats.RootCAs()`, will use TLS connection, but will be an anonymous one. This is the minimum.

Indeed, a good idea to test cert auth as well. Will implement.